designs/no-schedule-taint.md

```diff
@@ -0,0 +1,37 @@
+# Karpenter NoSchedule Taint [v0.32.0+]
+
+## Overview
+
+Karpenter uses the `node.kubernetes.io/unschedulable:NoSchedule` taint to prevent pods from scheduling to nodes being deprovisioned. The `node.kubernetes.io/unschedulable` taint is well-known in Kubernetes and may be leveraged or relied on by other controllers or applications.
+
+Karpenter cordons nodes when it launching replacements for a deprovisioning action. When Karpenter crashes or restarts during a deprovisioning action, nodes can be left cordoned until manual intervention or deprovisioning possibly through Karpenter in the future. Karpenter can't distinguish between nodes that it has cordoned and nodes a user/another controller has cordoned. **Since Karpenter cannot assume it’s the only agent in the cluster managing this taint, Karpenter is unable to recover from crashes where nodes are left cordoned.**
+
+Karpenter is [increasing the parallelism of deprovisioning](https://github.com/aws/karpenter-core/pull/542), allowing Karpenter to execute many actions simultaneously. Once merged, Karpenter could have many actions in memory, making it likely for a Karpenter crash to have more widespread and noticeable effects. At the worst, a cluster could have all nodes being deprovisioned simultaneously, where a crash would result in all nodes being cordoned, requiring new nodes for any new pods.
+
+PR: https://github.com/aws/karpenter-core/pull/508
```

Comment:
Do you want this PR linked in the overview section? I'm not sure which section it fits in, but it seems odd for it to be in the overview block

Reply:
Good callout, moved it to Proposal.